Extension/src/LanguageServer/cppBuildTaskProvider.ts

```diff
@@ -445,7 +445,7 @@ class CustomBuildTaskTerminal implements Pseudoterminal {
 
     private printBuildSummary(error: string, stdout: string, stderr: string): number {
         if (error || (!stdout && stderr && stderr.includes("error")) ||
-            (stdout && stdout.includes("error C"))) { // cl.exe compiler errors
+            (stdout && (stdout.includes("error C") || stdout.includes("LINK : fatal error")))) { // cl.exe compiler errors
```

Comment:
I'm not sure we'll be able to reliably cover all possible failure strings (including localized compilers) here.  Fixing the Promise to return the result instead of 0 and checking it here would be the more robust solution.  Can you pass in the error code to this function and check it as well?  You may be able to remove the other string checks.

Reply:
This section of code is for cl.exe errors and cl.exe messages aren't localized, right? Your suggestion might still be a better approach, I haven't checked.

Reply:
This checks for both result value and the stdout

Reply:
> This checks for both the result value and the stdout string

I don't see it.  `printBuildSummary` only accepts an error string, stdout, and stderr.  It also needs to look at the value of `spawnResult` from line 409.

Reply:
that is right, I only considered it in firing the results.